apps/frontend/src/modules/auth/slices/actions.ts

```diff
@@ -14,10 +14,14 @@ const signUp = createAsyncThunk<
 	UserSignUpResponseDto,
 	UserSignUpRequestDto,
 	AsyncThunkConfig
->(`${sliceName}/sign-up`, (registerPayload, { extra }) => {
-	const { authApi } = extra;
+>(`${sliceName}/sign-up`, async (registerPayload, { extra }) => {
+	const { authApi, storage } = extra;
+
+	const { token, user } = await authApi.signUp(registerPayload);
+
+	await storage.set(StorageKey.TOKEN, token);
 
-	return authApi.signUp(registerPayload);
+	return { token, user };
```

Comment:
```suggestion
	return user;
```
not sure that we need to pass a token somewhere